frame/collective/src/lib.rs

```diff
@@ -1441,6 +1458,186 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn motions_all_first_vote_free_works() {
+		new_test_ext().execute_with(|| {
+			let proposal = make_proposal(42);
+			let proposal_len: u32 = proposal.using_encoded(|p| p.len() as u32);
+			let hash: H256 = proposal.blake2_256().into();
+			let end = 4;
+			assert_ok!(Collective::propose(Origin::signed(1), 2, Box::new(proposal.clone()), proposal_len));
+			assert_eq!(
+				Collective::voting(&hash),
+				Some(Votes { index: 0, threshold: 2, ayes: vec![1], nays: vec![], end })
+			);
+
+			// For the motion, acc 2's first vote, expecting Ok with Pays::No ...
+			assert_eq!( Collective::vote(Origin::signed(2), hash.clone(), 0, true),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							207711000,
+						),
+						pays_fee: Pays::No,
+					}
+				)
+			);
+
+			// Duplicate vote, expecting error with Pays::Yes ...
+			let vote_rval: DispatchResultWithPostInfo = Collective::vote(Origin::signed(2), hash.clone(), 0, true);
+			match vote_rval {
+				Ok(_) => {
+					println!( "@[{:#?}::{:#?}]::vote-fee() | Should not Occur",
+						file!(),
+						line!() );
+				},
```

Comment:
This should panic, forcing the test to fail.

Reply:
Thanks Shawn, replaced with suggested one.

![](https://i.imgur.com/lfd2Bwo.png)